pkg/agent/route/route_linux.go

```diff
@@ -365,21 +365,25 @@ func getLocalAntreaFlexibleIPAMPodIPSetName(isIPv6 bool) string {
 	}
 }
 
-// writeEKSMangleRule writes an additional iptables mangle rule to the
-// iptablesData buffer, which is required to ensure that the reverse path for
-// NodePort Service traffic is correct on EKS.
-// See https://github.com/antrea-io/antrea/issues/678.
-func (c *Client) writeEKSMangleRule(iptablesData *bytes.Buffer) {
+// writeEKSMangleRules writes an additional iptables mangle rule to the
+// iptablesData buffer, to set the traffic mark to the connection mark for
+// traffic coming out of the gateway. This rule is needed for 2 cases:
+//  * for the reverse path for NodePort Service traffic (see
+//    https://github.com/antrea-io/antrea/issues/678).
+//  * for Pod-to-external traffic that needs to be SNATed (see
+//    https://github.com/antrea-io/antrea/issues/3946).
+func (c *Client) writeEKSMangleRules(iptablesData *bytes.Buffer) {
 	// TODO: the following should be taking into account:
-	//   1) AWS_VPC_CNI_NODE_PORT_SUPPORT may be set to false (by default is
-	//   true), in which case we do not need to install the rule.
+	//   1) this rule is only needed if AWS_VPC_CNI_NODE_PORT_SUPPORT is set
+	//   to true (which is the default) or if AWS_VPC_K8S_CNI_EXTERNALSNAT
+	//   is set to false (which is also the default). If both settings are
+	//   changed, we do not need to install the rule.
 	//   2) this option is not documented but the mark value can be
 	//   configured with AWS_VPC_K8S_CNI_CONNMARK.
-	// We could look for the rule added by AWS VPC CNI to the mangle
-	// table. If it does not exist, we do not need to install this rule. If
-	// it does exist we can scan for the mark value and use that in our
-	// rule.
-	klog.V(2).Infof("Add iptable mangle rule for EKS to ensure correct reverse path for NodePort Service traffic")
+	// While we do not have access to these environment variables, we could
+	// look for existing rules installed by the AWS VPC CNI, and determine
+	// whether we need to install this rule.
```

Comment:
Is this still true? I didn't find this code.

Reply:
This is part of the TODO paragraph. We don't implement this as the moment, we assume that `aws-node` (the AWS DaemonSet) is running with default configuration, which is likely to be the case for 99% of users.